docs/superpowers/specs/2026-05-18-recorder-vocab-distribution-design.md

```
@@ -0,0 +1,205 @@
+    vocab shape  (80 buckets over id 0..151935, log-y)
+    ▇▇▇▅▅▄▄▃▃▃▂▂▂▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▁ ▁ ▁  ▁  ▁  ▁         ▁       ▁    ▁       ▁
+    0                            38K                          76K                       114K                152K
+
+    min_tokens  mean    32.0   p50    32
+    ignore_eos=true: 41
+```
+
+Rules:
+- The blank lines between every block are added unconditionally as part of this change.
+- The vocab block is rendered only when `vocab_distribution` is non-null. When it's null (no requests on the endpoint), neither the `Vocab` line nor the `vocab shape` block prints.
+- `top:` line shows the first 5 entries of `top_tokens` formatted as `"<text>" <count>` joined with `, `. The text is wrapped in literal double quotes with any leading/trailing whitespace preserved (e.g. `" the" 3201`). If `text` contains a non-printable character or `tokenizer.decode([id])` raised, render that entry as `<id=N> <count>` instead (no quotes).
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)